File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import *

def solution(people, limit):
    answer = 0
    start = 0
    end = len(people) - 1
    flag = True
    
    people.sort()
    people = deque(people)

    while end > start:
        if people[start] + people[end] > limit:
            end -= 1
            answer += 1

        else:
            answer += 1
            start += 1
            end -= 1
            
    if end == start:
        answer += 1
        
    return answer

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        /*
        구명보트 1개에는 최대 100kg, 최대한 적은 구명보트를 사용해야 함.
        people의 최대 수는 50,000 O(N^2) 이하로 ㄱㄱ
        
        
        */
        int answer = 0;
        int i = 0;
        int j = people.length - 1;
        Arrays.sort(people);
        
        while (i <= j) {
            if (people[i] + people[j] <= limit) {
                i++;
            }
            
            j -= 1;
            answer++;
        }
        
        return answer;
    }
}